river: init at unstable-2021-04-08

heavyrain266 opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
Hi, I just wanted to add new Wayland Compositor into master branch, it's quite popular but just saw that a lot of people have problems with building/installing it on nixos. I have tested this derivation by building it through selhosted hydra, locally on nixos with nix-build and on non-nixos distribution locally and from own channel. I'm not sure about versioning in commit message since they doesn't provide any github releases yet.

Here is github page for this package https://github.com/ifreund/river

###### Things done 

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

heavyrain266 committed:
river: init at master

river: init at unstable 2021-04-08

river: init at unstable 2021-04-08

Co-authored-by: fortuneteller2k <20619776+fortuneteller2k@users.noreply.github.com>

nixinator commented:
is there anyway you get them to make a tagged or release, sometimes if you ask nicely...they will :-).

branwright1 commented:
> is there anyway you get them to make a tag/release, sometimes if you ask nicely...they will :-).

I will ask them later today, haven't thinked about it.

fortuneteller2k commented:
> is there anyway you get them to make a tag/release, sometimes if you ask nicely...they will :-).

If this isn't possible, just make the version `unstable-yyyy-mm-dd` with the date of the commit.

fortuneteller2k commented on pkgs/applications/window-managers/river/default.nix:
```diff
@@ -0,0 +1,39 @@
+
+  meta = with lib; {
+    description = "A dynamic tiling wayland compositor";
+    longDescriotion = ''
```
```suggestion
    longDescription = ''
```

fortuneteller2k commented on pkgs/applications/window-managers/river/default.nix:
```diff
@@ -0,0 +1,39 @@
+  preBuild = "export HOME=$TMPDIR;";
+  installPhase = "zig build -Drelease-safe -Dxwayland -Dman-pages --prefix $out install";
+
+  nativeBuildInputs = [ zig scdoc pkgconfig ];
```
```suggestion
  nativeBuildInputs = [ zig scdoc pkg-config ];
```

fortuneteller2k replied:
`pkgconfig` is an alias for `pkg-config`

fortuneteller2k commented on pkgs/applications/window-managers/river/default.nix:
```diff
@@ -0,0 +1,39 @@
+{ lib, stdenv ,fetchFromGitHub
+, zig, pkgconfig, scdoc
```
```suggestion
, zig, pkg-config, scdoc
```

fortuneteller2k replied:
`pkgconfig` is an alias for `pkg-config`

fortuneteller2k commented on pkgs/applications/window-managers/river/default.nix:
```diff
@@ -0,0 +1,39 @@
+    homepage = "https://github.com/ifreund/river";
+    license = licenses.gpl3Plus;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ branwright1 ];
```
Add yourself to the [maintainer list](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix) with a separate commit `maintainers: add branwright1`

fortuneteller2k commented on pkgs/applications/window-managers/river/default.nix:
```diff
@@ -0,0 +1,39 @@
+}:
+
+stdenv.mkDerivation rec {
+  name = "river";
```
```suggestion
  pname = "river";
  version = "unstable-2021-04-08";
```

fortuneteller2k commented on pkgs/applications/window-managers/river/default.nix:
```diff
@@ -0,0 +1,39 @@
+  meta = with lib; {
+    description = "A dynamic tiling wayland compositor";
+    longDescription = ''
+    river is a dynamic tiling wayland compositor that takes inspiration from dwm and bspwm. 
```
```suggestion
    river is a dynamic tiling wayland compositor that takes inspiration from dwm and bspwm.
```

heavyrain266 committed:
maintainers: add branwright1

river: init at unstable 2021-04-08

river: init at unstable 2021-04-08

river: init at unstable 2021-04-08

river: init at unstable 2021-04-08

river: init at unstable 2021-04-08

river: init at unstable 2021-04-08

river: init at unstable 2021-04-08

branwright1 commented:
Ok, changed river commit to unstable 2021-04-06 from master and added myself to maintainers.

fortuneteller2k commented:
Add a `river` entry in the [top-level package list](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix).

```nix
river = callPackage ../applications/window-managers/river { };
```

branwright1 commented:
Ofbord failed after adding callPackages to `all-packages.nix` because libudev is restricted for trusted users

fortuneteller2k commented:
Result of `nixpkgs-review pr 119387` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>river</li>
  </ul>
</details>

fortuneteller2k commented on pkgs/applications/window-managers/river/default.nix:
```diff
@@ -0,0 +1,39 @@
+  };
+
+  buildInputs = [ wayland xwayland wayland-protocols wlroots pixman
+    libxkbcommon pixman libudev libevdev libX11 libGL
```
```suggestion
    libxkbcommon pixman udev libevdev libX11 libGL
```

fortuneteller2k replied:
`libudev` is an alias for `udev`

fortuneteller2k commented on pkgs/applications/window-managers/river/default.nix:
```diff
@@ -0,0 +1,39 @@
+{ lib, stdenv ,fetchFromGitHub
+, zig, pkg-config, scdoc
+, wayland ,xwayland, wayland-protocols, wlroots
+, libxkbcommon, pixman, libudev, libevdev, libX11, libGL
```
```suggestion
, libxkbcommon, pixman, udev, libevdev, libX11, libGL
```

fortuneteller2k replied:
`libudev` is an alias for `udev`

branwright1 commented:
How do I know which packages are aliasses?

fortuneteller2k commented:
> How do I know which packages are aliasses?

Check this [file](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/aliases.nix)

moni-dz reviewed (approved):
Diff LGTM.

fortuneteller2k commented:
Result of `nixpkgs-review pr 119387` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>river</li>
  </ul>
</details>

SuperSandro2000 commented on pkgs/applications/window-managers/river/default.nix:
```diff
@@ -0,0 +1,40 @@
+    fetchSubmodules = true;
+  };
+
+  buildInputs = [ wayland xwayland wayland-protocols wlroots pixman
```
Is a binary from wayland executed at build time? If so it needs to be in nativeBuildInouts.

branwright1 replied:
Wayland is xml file, there is no binary for it, I think it's runtime dependencie

SuperSandro2000 commented on pkgs/applications/window-managers/river/default.nix:
```diff
@@ -0,0 +1,40 @@
+    libxkbcommon pixman udev libevdev libX11 libGL
+  ];
+
+  preBuild = "export HOME=$TMPDIR;";
+  installPhase = "zig build -Drelease-safe -Dxwayland -Dman-pages --prefix $out install";
```
```suggestion
  preBuild = ''
    export HOME=$TMPDIR
  '';

  installPhase = ''
    zig build -Drelease-safe -Dxwayland -Dman-pages --prefix $out install
  '';
```

SuperSandro2000 commented on pkgs/applications/window-managers/river/default.nix:
```diff
@@ -0,0 +1,40 @@
+  meta = with lib; {
+    description = "A dynamic tiling wayland compositor";
+    longDescription = ''
+    river is a dynamic tiling wayland compositor that takes inspiration from dwm and bspwm.
```
```suggestion
      river is a dynamic tiling wayland compositor that takes inspiration from dwm and bspwm.
```

branwright1 commented:
@SuperSandro2000 Can you check if it's ok now? commited and squashed those changes.
